### PR TITLE
1.0.0 release candidate.

### PR DESCRIPTION
**Pull request by opi-smccoole**
Added support for the following:

1. Add and use `get_block_info` to provide a lighter weight way to calculate TAPOS instead of a full `get_block` call.
2. Add and use `send_transaction` in place of `push_transaction`.
3. Add ability to use last irreversible block for TAPOS and made it the default.  User can configure to use blocks behind.
4. Add support for action return values.
5. Add support for `get_kv_table_rows`.
6. Update unit tests.
7. Update README and CONTRIBUTING.
8. Add EXAMPLES file for more advanced snippets on SDK use.
9. Update javadocs.

### CODE REVIEW

File: eosiojava/src/test/java/one/block/eosiojava/session/NegativeTransactionProcessorTest.java [hunk 1]
```diff
@@ -14,10 +14,7 @@
 import one.block.eosiojava.error.EosioError;
 import one.block.eosiojava.error.ErrorConstants;
 import one.block.eosiojava.error.abiProvider.GetAbiError;
-import one.block.eosiojava.error.rpcProvider.GetBlockRpcError;
-import one.block.eosiojava.error.rpcProvider.GetInfoRpcError;
-import one.block.eosiojava.error.rpcProvider.GetRequiredKeysRpcError;
-import one.block.eosiojava.error.rpcProvider.PushTransactionRpcError;
+import one.block.eosiojava.error.rpcProvider.*;
```
Comment (DanteNguyen1008) [inline eosiojava/src/test/java/one/block/eosiojava/session/NegativeTransactionProcessorTest.java:17]: please avoid wildcard import

File: eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java [hunk 2]
```diff
@@ -14,10 +14,7 @@
 import java.util.Arrays;
 import java.util.List;
 import one.block.eosiojava.error.abiProvider.GetAbiError;
-import one.block.eosiojava.error.rpcProvider.GetBlockRpcError;
-import one.block.eosiojava.error.rpcProvider.GetInfoRpcError;
-import one.block.eosiojava.error.rpcProvider.GetRequiredKeysRpcError;
-import one.block.eosiojava.error.rpcProvider.PushTransactionRpcError;
+import one.block.eosiojava.error.rpcProvider.*;
```
Comment (DanteNguyen1008) [inline eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java:17]: please avoid using wildcard import

File: eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java [hunk 3]
```diff
@@ -41,13 +38,11 @@
...
+import one.block.eosiojava.models.rpcProvider.request.GetBlockInfoRequest;
 import one.block.eosiojava.models.rpcProvider.request.GetBlockRequest;
 import one.block.eosiojava.models.rpcProvider.request.GetRequiredKeysRequest;
-import one.block.eosiojava.models.rpcProvider.request.PushTransactionRequest;
-import one.block.eosiojava.models.rpcProvider.response.GetBlockResponse;
-import one.block.eosiojava.models.rpcProvider.response.GetInfoResponse;
-import one.block.eosiojava.models.rpcProvider.response.GetRequiredKeysResponse;
-import one.block.eosiojava.models.rpcProvider.response.PushTransactionResponse;
+import one.block.eosiojava.models.rpcProvider.request.SendTransactionRequest;
+import one.block.eosiojava.models.rpcProvider.response.*;
```
Comment (DanteNguyen1008) [inline eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java:45]: please avoid using wildcard import

**Review by DanteNguyen1008 — COMMENTED** [review]
I just have some minor comments. All looks good.

File: eosiojava/src/main/java/one/block/eosiojava/interfaces/IRPCProvider.java [hunk 4]
```diff
@@ -1,19 +1,8 @@
...
-import one.block.eosiojava.models.rpcProvider.request.GetBlockRequest;
-import one.block.eosiojava.models.rpcProvider.request.GetRawAbiRequest;
-import one.block.eosiojava.models.rpcProvider.request.GetRequiredKeysRequest;
-import one.block.eosiojava.models.rpcProvider.request.PushTransactionRequest;
-import one.block.eosiojava.models.rpcProvider.response.GetBlockResponse;
-import one.block.eosiojava.models.rpcProvider.response.GetInfoResponse;
-import one.block.eosiojava.models.rpcProvider.response.GetRawAbiResponse;
-import one.block.eosiojava.models.rpcProvider.response.GetRequiredKeysResponse;
-import one.block.eosiojava.models.rpcProvider.response.PushTransactionResponse;
+import one.block.eosiojava.error.rpcProvider.*;
```
[the remark this answers is not in the document]
Comment (opi-smccoole) [reply eosiojava/src/main/java/one/block/eosiojava/interfaces/IRPCProvider.java:3]: I had thought I had turned that off but I guess there is an extra step:

Disable wildcard imports to always import single classes﻿
In the Settings/Preferences dialog ⌘,, select Code Style | Java | Imports.

Make sure that the Use single class import option is enabled.

In the Class count to use import with ‘*’ and Names count to use static import with ‘*’ fields, specify values that definitely exceed the number of classes in a package and the number of names in a class (for example, 999).

I didn't do the last one.  Will fix.

File: eosiojava/src/main/java/one/block/eosiojava/interfaces/ISerializationProvider.java [hunk 5]
```diff
@@ -1,11 +1,6 @@
 package one.block.eosiojava.interfaces;
 
-import one.block.eosiojava.error.serializationProvider.DeserializeAbiError;
-import one.block.eosiojava.error.serializationProvider.DeserializeError;
-import one.block.eosiojava.error.serializationProvider.DeserializeTransactionError;
-import one.block.eosiojava.error.serializationProvider.SerializeAbiError;
-import one.block.eosiojava.error.serializationProvider.SerializeError;
-import one.block.eosiojava.error.serializationProvider.SerializeTransactionError;
+import one.block.eosiojava.error.serializationProvider.*;
```
[the remark this answers is not in the document]
Comment (opi-smccoole) [reply eosiojava/src/main/java/one/block/eosiojava/interfaces/ISerializationProvider.java]: Fixed

File: eosiojava/src/main/java/one/block/eosiojava/models/rpcProvider/response/SendTransactionResponse.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     * submitted.
+     *
+     * @return ArrayList of Objects containing the return values from the response.
+     */
+    public ArrayList<Object> getActionValues() {
+        ArrayList<Object> returnValues = new ArrayList<Object>();
+        if (processed == null) { return returnValues; }
+        if (!processed.containsKey("action_traces")) { return returnValues; }
+        for (Map trace : (List<Map>) processed.get("action_traces")) {
+            returnValues.add(trace.getOrDefault("return_value_data", null));
```
[the remark this answers is not in the document]
Comment (opi-smccoole) [reply eosiojava/src/main/java/one/block/eosiojava/models/rpcProvider/response/SendTransactionResponse.java]: Added a constant field.

File: eosiojava/src/main/java/one/block/eosiojava/models/rpcProvider/response/SendTransactionResponse.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+     * There may be more action values than input actions due to inline actions or notifications
+     * but input (request) actions are always returned first and in the same order as they were
+     * submitted.
+     *
+     * @return ArrayList of Objects containing the return values from the response.
+     */
+    public ArrayList<Object> getActionValues() {
+        ArrayList<Object> returnValues = new ArrayList<Object>();
+        if (processed == null) { return returnValues; }
+        if (!processed.containsKey("action_traces")) { return returnValues; }
```
[the remark this answers is not in the document]
Comment (opi-smccoole) [reply eosiojava/src/main/java/one/block/eosiojava/models/rpcProvider/response/SendTransactionResponse.java]: Added a constant field.

File: eosiojava/src/test/java/one/block/eosiojava/session/NegativeTransactionProcessorTest.java (hunk 1 above)
Comment (opi-smccoole) [reply eosiojava/src/test/java/one/block/eosiojava/session/NegativeTransactionProcessorTest.java]: Fixed.

File: eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java (hunk 3 above)
Comment (opi-smccoole) [reply eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java:47]: Fixed

File: eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java (hunk 2 above)
Comment (opi-smccoole) [reply eosiojava/src/test/java/one/block/eosiojava/session/TransactionProcessorTest.java]: fixed

File: eosiojava/src/main/java/one/block/eosiojava/session/TransactionProcessor.java [hunk 8]
```diff
@@ -9,12 +9,10 @@
 import java.util.List;
 import one.block.eosiojava.error.ErrorConstants;
 import one.block.eosiojava.error.abiProvider.GetAbiError;
-import one.block.eosiojava.error.rpcProvider.GetBlockRpcError;
-import one.block.eosiojava.error.rpcProvider.GetInfoRpcError;
-import one.block.eosiojava.error.rpcProvider.GetRequiredKeysRpcError;
-import one.block.eosiojava.error.rpcProvider.PushTransactionRpcError;
+import one.block.eosiojava.error.rpcProvider.*;
```
[the remark this answers is not in the document]
Comment (opi-smccoole) [reply eosiojava/src/main/java/one/block/eosiojava/session/TransactionProcessor.java]: fixed

**Review by DanteNguyen1008 — APPROVED** [review]

**Review by GonnaGitYou — APPROVED** [review]

**Pull request merged**